app/javascript/components/visualization/ScatterPlot.jsx

```diff
@@ -370,72 +368,97 @@ function RawScatterPlot({
+          if (imageCacheHit) {
+            await drawPlotImage(response)
+          }
 
-      fetchBucketFile(bucketId, imagePath).then(async response => {
-        const imageCacheHit = response.ok
+          // Then make it interactive, using gene expression scatter plot data array from GCS bucket
+          await renderBucketData(fetchMethod, expressionParams, dataPath)
 
-        // Draw plot as static image first, if it's cached
-        if (imageCacheHit) {
-          await drawPlotImage(response)
+          // TODO (SCP-4839): Instrument more bucket cache analytics, then remove console log below
+          // console.log(`Image render took ${ Date.now() - window.t0}`)
+          // Add imageCacheHit boolean to perfTime object here
+        })
+      } else {
+        try {
+          // attempt to fetch the data, this will use the cache if available
+          const respData1 = await fetchMethod({
+            studyAccession,
+            cluster,
+            annotation: annotation ? annotation : '',
+            subsample,
+            consensus,
+            genes,
+            isAnnotatedScatter,
+            isCorrelatedScatter,
+            expressionArray
+          })
+          // check that the data contains annotations needed for processing scatterplot
+          if (respData1[0]?.data?.annotations?.length) {
+            processScatterPlot(respData1)
+          } else {
+            // if the data was missing the necessary info, make an api call
+            const respData = await fetchCluster({
+              studyAccession,
+              cluster,
+              annotation: annotation ? annotation : '',
+              subsample,
+              consensus,
+              genes,
+              isAnnotatedScatter,
+              isCorrelatedScatter,
+              expressionArray
+            })
+            processScatterPlot(respData)
+          }
+        } catch (error) {
+          setIsLoading(false)
+          setShowError(true)
+          setError(error)
         }
-
-        // Then make it interactive, using gene expression scatter plot data array from GCS bucket
-        renderBucketData(fetchMethod, expressionParams, dataPath)
-
-        // TODO (SCP-4839): Instrument more bucket cache analytics, then remove console log below
-        // console.log(`Image render took ${ Date.now() - window.t0}`)
-        // Add imageCacheHit boolean to perfTime object here
-      })
-    } else {
-      fetchMethod({
-        studyAccession,
-        cluster,
-        annotation: annotation ? annotation : '',
-        subsample,
-        consensus,
-        genes,
-        isAnnotatedScatter,
-        isCorrelatedScatter,
-        expressionArray
-      }).then(processScatterPlot).catch(error => {
-        setIsLoading(false)
-        setShowError(true)
-        setError(error)
-      })
+      }
     }
-  }, [cluster, annotation.name, subsample, consensus, genes.join(','), isAnnotatedScatter])
+    getData()
```

Comment:
SCP JS conventionally includes "fetch" in a function name when it involves sending network requests for something (and when it might be cached).  This helps distinguish at a glance that the function doesn't simply look up a value from memory.

```suggestion
    fetchData()
```